{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c153bfa7-b139-480f-86a6-07ceb803c6d9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "    <div class=\"bk-root\">\n",
       "        <a href=\"https://bokeh.org\" target=\"_blank\" class=\"bk-logo bk-logo-small bk-logo-notebook\"></a>\n",
       "        <span id=\"1002\">Loading BokehJS ...</span>\n",
       "    </div>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "\n",
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  const force = true;\n",
       "\n",
       "  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "  const JS_MIME_TYPE = 'application/javascript';\n",
       "  const HTML_MIME_TYPE = 'text/html';\n",
       "  const EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  const CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    const script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    const cell = handle.cell;\n",
       "\n",
       "    const id = cell.output_area._bokeh_element_id;\n",
       "    const server_id = cell.output_area._bokeh_server_id;\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null && id in Bokeh.index) {\n",
       "      Bokeh.index[id].model.document.clear();\n",
       "      delete Bokeh.index[id];\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      const cmd_clean = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd_clean, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            const id = msg.content.text.trim();\n",
       "            if (id in Bokeh.index) {\n",
       "              Bokeh.index[id].model.document.clear();\n",
       "              delete Bokeh.index[id];\n",
       "            }\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      const cmd_destroy = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd_destroy);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    const output_area = handle.output_area;\n",
       "    const output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!Object.prototype.hasOwnProperty.call(output.data, EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    const toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      const bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      const script_attrs = bk_div.children[0].attributes;\n",
       "      for (let i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      const toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      const props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    const events = require('base/js/events');\n",
       "    const OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  \n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  const NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    const el = document.getElementById(\"1002\");\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) {\n",
       "        if (callback != null)\n",
       "          callback();\n",
       "      });\n",
       "    } finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.debug(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(css_urls, js_urls, callback) {\n",
       "    if (css_urls == null) css_urls = [];\n",
       "    if (js_urls == null) js_urls = [];\n",
       "\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = css_urls.length + js_urls.length;\n",
       "\n",
       "    function on_load() {\n",
       "      root._bokeh_is_loading--;\n",
       "      if (root._bokeh_is_loading === 0) {\n",
       "        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n",
       "        run_callbacks()\n",
       "      }\n",
       "    }\n",
       "\n",
       "    function on_error(url) {\n",
       "      console.error(\"failed to load \" + url);\n",
       "    }\n",
       "\n",
       "    for (let i = 0; i < css_urls.length; i++) {\n",
       "      const url = css_urls[i];\n",
       "      const element = document.createElement(\"link\");\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error.bind(null, url);\n",
       "      element.rel = \"stylesheet\";\n",
       "      element.type = \"text/css\";\n",
       "      element.href = url;\n",
       "      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n",
       "      document.body.appendChild(element);\n",
       "    }\n",
       "\n",
       "    for (let i = 0; i < js_urls.length; i++) {\n",
       "      const url = js_urls[i];\n",
       "      const element = document.createElement('script');\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error.bind(null, url);\n",
       "      element.async = false;\n",
       "      element.src = url;\n",
       "      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.head.appendChild(element);\n",
       "    }\n",
       "  };\n",
       "\n",
       "  function inject_raw_css(css) {\n",
       "    const element = document.createElement(\"style\");\n",
       "    element.appendChild(document.createTextNode(css));\n",
       "    document.body.appendChild(element);\n",
       "  }\n",
       "\n",
       "  \n",
       "  const js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-2.4.2.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.4.2.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.4.2.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.4.2.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-2.4.2.min.js\"];\n",
       "  const css_urls = [];\n",
       "  \n",
       "\n",
       "  const inline_js = [\n",
       "    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "    function(Bokeh) {\n",
       "    \n",
       "    \n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    \n",
       "    if (root.Bokeh !== undefined || force === true) {\n",
       "      \n",
       "    for (let i = 0; i < inline_js.length; i++) {\n",
       "      inline_js[i].call(root, root.Bokeh);\n",
       "    }\n",
       "    if (force === true) {\n",
       "        display_loaded();\n",
       "      }} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      const cell = $(document.getElementById(\"1002\")).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(css_urls, js_urls, function() {\n",
       "      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  const force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n  \n\n  \n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  const NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    const el = document.getElementById(\"1002\");\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error(url) {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (let i = 0; i < css_urls.length; i++) {\n      const url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    for (let i = 0; i < js_urls.length; i++) {\n      const url = js_urls[i];\n      const element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.async = false;\n      element.src = url;\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  \n  const js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-2.4.2.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.4.2.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.4.2.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.4.2.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-2.4.2.min.js\"];\n  const css_urls = [];\n  \n\n  const inline_js = [\n    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\n    function(Bokeh) {\n    \n    \n    }\n  ];\n\n  function run_inline_js() {\n    \n    if (root.Bokeh !== undefined || force === true) {\n      \n    for (let i = 0; i < inline_js.length; i++) {\n      inline_js[i].call(root, root.Bokeh);\n    }\n    if (force === true) {\n        display_loaded();\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      const cell = $(document.getElementById(\"1002\")).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "\n",
    "import bokeh.io\n",
    "import bokeh.plotting\n",
    "bokeh.io.output_notebook()\n",
    "\n",
    "plt.rcParams['figure.dpi'] = 150\n",
    "import glob, os, json"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d3f070eb-6ec0-4438-b74f-c18093aaa5d3",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "1865f746-daaf-4300-9507-ef39ee0062e7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2085"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "486+487+488+131+493"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "57432abb-dc9c-4c6b-b59c-ed644de01112",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8c020e1c-9dd2-4f1d-88fc-fc03c5fea0bb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "42e34afe-0208-4063-8917-b06dedd157a6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "37d9b676-cfec-4b77-aeb7-abdf25700e14",
   "metadata": {},
   "source": [
    "# Virus Intact Database\n",
    "\n",
    "42,154 pairwise interactions from <a href=\"https://www.ebi.ac.uk/legacy-intact/query/annot:%22dataset:virus%22?conversationContext=7\" target=\"\\blank\">here</a>."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "561af12d-215c-4fe1-bce1-c1b11f33f98a",
   "metadata": {},
   "outputs": [],
   "source": [
    "human_human = pd.read_csv(\"Processed/human_important_for_virus.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "1a993a1b-43f0-4d46-9d7b-bb38a7767811",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(26107, 14)\n",
      "(26045, 14)\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "9428c92f-49fa-4a0b-b066-13501da8e12a",
   "metadata": {},
   "outputs": [],
   "source": [
    "test = pd.read_csv(\"Data/human_genes_all/genage_human.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "a196fa03-0e35-4171-b932-62c4a13db6f2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GenAge ID</th>\n",
       "      <th>symbol</th>\n",
       "      <th>name</th>\n",
       "      <th>entrez gene id</th>\n",
       "      <th>uniprot</th>\n",
       "      <th>why</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>GHR</td>\n",
       "      <td>growth hormone receptor</td>\n",
       "      <td>2690</td>\n",
       "      <td>GHR_HUMAN</td>\n",
       "      <td>mammal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>GHRH</td>\n",
       "      <td>growth hormone releasing hormone</td>\n",
       "      <td>2691</td>\n",
       "      <td>SLIB_HUMAN</td>\n",
       "      <td>mammal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>SHC1</td>\n",
       "      <td>SHC (Src homology 2 domain containing) transfo...</td>\n",
       "      <td>6464</td>\n",
       "      <td>SHC1_HUMAN</td>\n",
       "      <td>mammal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>POU1F1</td>\n",
       "      <td>POU class 1 homeobox 1</td>\n",
       "      <td>5449</td>\n",
       "      <td>PIT1_HUMAN</td>\n",
       "      <td>mammal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>PROP1</td>\n",
       "      <td>PROP paired-like homeobox 1</td>\n",
       "      <td>5626</td>\n",
       "      <td>PROP1_HUMAN</td>\n",
       "      <td>mammal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>302</th>\n",
       "      <td>305</td>\n",
       "      <td>TRAP1</td>\n",
       "      <td>TNF receptor-associated protein 1</td>\n",
       "      <td>10131</td>\n",
       "      <td>TRAP1_HUMAN</td>\n",
       "      <td>functional,putative</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303</th>\n",
       "      <td>306</td>\n",
       "      <td>TRPV1</td>\n",
       "      <td>transient receptor potential cation channel su...</td>\n",
       "      <td>7442</td>\n",
       "      <td>TRPV1_HUMAN</td>\n",
       "      <td>mammal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304</th>\n",
       "      <td>307</td>\n",
       "      <td>NFE2L1</td>\n",
       "      <td>nuclear factor, erythroid 2-like 1</td>\n",
       "      <td>4779</td>\n",
       "      <td>NF2L1_HUMAN</td>\n",
       "      <td>model,functional</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>305</th>\n",
       "      <td>308</td>\n",
       "      <td>IFNB1</td>\n",
       "      <td>Interferon beta</td>\n",
       "      <td>3456</td>\n",
       "      <td>IFNB_HUMAN</td>\n",
       "      <td>mammal,cell</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306</th>\n",
       "      <td>309</td>\n",
       "      <td>GDF11</td>\n",
       "      <td>growth differentiation factor 11</td>\n",
       "      <td>10220</td>\n",
       "      <td>A0A024RB20_HUMAN</td>\n",
       "      <td>mammal,human_link</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>307 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     GenAge ID  symbol                                               name  \\\n",
       "0            1     GHR                            growth hormone receptor   \n",
       "1            2    GHRH                   growth hormone releasing hormone   \n",
       "2            3    SHC1  SHC (Src homology 2 domain containing) transfo...   \n",
       "3            4  POU1F1                             POU class 1 homeobox 1   \n",
       "4            5   PROP1                        PROP paired-like homeobox 1   \n",
       "..         ...     ...                                                ...   \n",
       "302        305   TRAP1                  TNF receptor-associated protein 1   \n",
       "303        306   TRPV1  transient receptor potential cation channel su...   \n",
       "304        307  NFE2L1                 nuclear factor, erythroid 2-like 1   \n",
       "305        308   IFNB1                                    Interferon beta   \n",
       "306        309   GDF11                   growth differentiation factor 11   \n",
       "\n",
       "     entrez gene id           uniprot                  why  \n",
       "0              2690         GHR_HUMAN               mammal  \n",
       "1              2691        SLIB_HUMAN               mammal  \n",
       "2              6464        SHC1_HUMAN               mammal  \n",
       "3              5449        PIT1_HUMAN               mammal  \n",
       "4              5626       PROP1_HUMAN               mammal  \n",
       "..              ...               ...                  ...  \n",
       "302           10131       TRAP1_HUMAN  functional,putative  \n",
       "303            7442       TRPV1_HUMAN               mammal  \n",
       "304            4779       NF2L1_HUMAN     model,functional  \n",
       "305            3456        IFNB_HUMAN          mammal,cell  \n",
       "306           10220  A0A024RB20_HUMAN    mammal,human_link  \n",
       "\n",
       "[307 rows x 6 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2743ea67-7bc1-4bca-ae51-cb0f96c2466b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "fba588e5-643f-46e3-8299-b6dcaea04b08",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>yourlist:M20220508F248CABF64506F29A91F8037F07B67D14C9C7EF</th>\n",
       "      <th>Entry</th>\n",
       "      <th>Entry name</th>\n",
       "      <th>Status</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Gene names</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Length</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>O00505</td>\n",
       "      <td>O00505</td>\n",
       "      <td>IMA4_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Importin subunit alpha-4 (Importin alpha Q2) (...</td>\n",
       "      <td>KPNA3 QIP2</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>521</td>\n",
       "      <td>MAENPSLENHRIKSFKNKGRDVETMRRHRNEVTVELRKNKRDEHLL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>O00629</td>\n",
       "      <td>O00629</td>\n",
       "      <td>IMA3_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Importin subunit alpha-3 (Importin alpha Q1) (...</td>\n",
       "      <td>KPNA4 QIP1</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>521</td>\n",
       "      <td>MADNEKLDNQRLKNFKNKGRDLETMRRQRNEVVVELRKNKRDEHLL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P52294</td>\n",
       "      <td>P52294</td>\n",
       "      <td>IMA5_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Importin subunit alpha-5 (Karyopherin subunit ...</td>\n",
       "      <td>KPNA1 RCH2</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>538</td>\n",
       "      <td>MTTPGKENFRLKSYKNKSLNPDEMRRRREEEGLQLRKQKREEQLFK...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P52292</td>\n",
       "      <td>P52292</td>\n",
       "      <td>IMA1_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Importin subunit alpha-1 (Karyopherin subunit ...</td>\n",
       "      <td>KPNA2 RCH1 SRP1</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>529</td>\n",
       "      <td>MSTNENANTPAARLHRFKNKGKDSTEMRRRRIEVNVELRKAKKDDQ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Q01196-1</td>\n",
       "      <td>Q01196</td>\n",
       "      <td>RUNX1_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Runt-related transcription factor 1 (Acute mye...</td>\n",
       "      <td>RUNX1 AML1 CBFA2</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>453</td>\n",
       "      <td>MRIPVDASTSRRFTPPSTALSPGKMSEALPLGAPDAGAALAGKLRS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7620</th>\n",
       "      <td>P13598</td>\n",
       "      <td>P13598</td>\n",
       "      <td>ICAM2_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Intercellular adhesion molecule 2 (ICAM-2) (CD...</td>\n",
       "      <td>ICAM2</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>275</td>\n",
       "      <td>MSSFGYRTLTVALFTLICCPGSDEKVFEVHVRPKKLAVEPKGSLEV...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7621</th>\n",
       "      <td>Q15782</td>\n",
       "      <td>Q15782</td>\n",
       "      <td>CH3L2_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Chitinase-3-like protein 2 (Chondrocyte protei...</td>\n",
       "      <td>CHI3L2</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>390</td>\n",
       "      <td>MGATTMDQKSLWAGVVVLLLLQGGSAYKLVCYFTNWSQDRQEPGKF...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7622</th>\n",
       "      <td>P43629</td>\n",
       "      <td>P43629</td>\n",
       "      <td>KI3L1_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Killer cell immunoglobulin-like receptor 3DL1 ...</td>\n",
       "      <td>KIR3DL1 CD158E NKAT3 NKB1</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>444</td>\n",
       "      <td>MSLMVVSMACVGLFLVQRAGPHMGGQDKPFLSAWPSAVVPRGGHVT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7623</th>\n",
       "      <td>Q9BUV8</td>\n",
       "      <td>Q9BUV8</td>\n",
       "      <td>RCAF1_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Respirasome Complex Assembly Factor 1 (Rab5-in...</td>\n",
       "      <td>RAB5IF C20orf24 RCAF1 PNAS-11</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>137</td>\n",
       "      <td>MSGGRRKEEPPQPQLANGALKVSVWSKVLRSDAAWEDKDEFLDVIY...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7624</th>\n",
       "      <td>Q9Y3P9</td>\n",
       "      <td>Q9Y3P9</td>\n",
       "      <td>RBGP1_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Rab GTPase-activating protein 1 (GAP and centr...</td>\n",
       "      <td>RABGAP1 HSPC094</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>1069</td>\n",
       "      <td>MDDKASVGKISVSSDSVSTLNSEDFVLVSRQGDETPSTNNGSDDEK...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7625 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     yourlist:M20220508F248CABF64506F29A91F8037F07B67D14C9C7EF   Entry  \\\n",
       "0                                                O00505         O00505   \n",
       "1                                                O00629         O00629   \n",
       "2                                                P52294         P52294   \n",
       "3                                                P52292         P52292   \n",
       "4                                              Q01196-1         Q01196   \n",
       "...                                                 ...            ...   \n",
       "7620                                             P13598         P13598   \n",
       "7621                                             Q15782         Q15782   \n",
       "7622                                             P43629         P43629   \n",
       "7623                                             Q9BUV8         Q9BUV8   \n",
       "7624                                             Q9Y3P9         Q9Y3P9   \n",
       "\n",
       "       Entry name    Status  \\\n",
       "0      IMA4_HUMAN  reviewed   \n",
       "1      IMA3_HUMAN  reviewed   \n",
       "2      IMA5_HUMAN  reviewed   \n",
       "3      IMA1_HUMAN  reviewed   \n",
       "4     RUNX1_HUMAN  reviewed   \n",
       "...           ...       ...   \n",
       "7620  ICAM2_HUMAN  reviewed   \n",
       "7621  CH3L2_HUMAN  reviewed   \n",
       "7622  KI3L1_HUMAN  reviewed   \n",
       "7623  RCAF1_HUMAN  reviewed   \n",
       "7624  RBGP1_HUMAN  reviewed   \n",
       "\n",
       "                                          Protein names  \\\n",
       "0     Importin subunit alpha-4 (Importin alpha Q2) (...   \n",
       "1     Importin subunit alpha-3 (Importin alpha Q1) (...   \n",
       "2     Importin subunit alpha-5 (Karyopherin subunit ...   \n",
       "3     Importin subunit alpha-1 (Karyopherin subunit ...   \n",
       "4     Runt-related transcription factor 1 (Acute mye...   \n",
       "...                                                 ...   \n",
       "7620  Intercellular adhesion molecule 2 (ICAM-2) (CD...   \n",
       "7621  Chitinase-3-like protein 2 (Chondrocyte protei...   \n",
       "7622  Killer cell immunoglobulin-like receptor 3DL1 ...   \n",
       "7623  Respirasome Complex Assembly Factor 1 (Rab5-in...   \n",
       "7624  Rab GTPase-activating protein 1 (GAP and centr...   \n",
       "\n",
       "                         Gene names              Organism  Length  \\\n",
       "0                        KPNA3 QIP2  Homo sapiens (Human)     521   \n",
       "1                        KPNA4 QIP1  Homo sapiens (Human)     521   \n",
       "2                        KPNA1 RCH2  Homo sapiens (Human)     538   \n",
       "3                   KPNA2 RCH1 SRP1  Homo sapiens (Human)     529   \n",
       "4                  RUNX1 AML1 CBFA2  Homo sapiens (Human)     453   \n",
       "...                             ...                   ...     ...   \n",
       "7620                          ICAM2  Homo sapiens (Human)     275   \n",
       "7621                         CHI3L2  Homo sapiens (Human)     390   \n",
       "7622      KIR3DL1 CD158E NKAT3 NKB1  Homo sapiens (Human)     444   \n",
       "7623  RAB5IF C20orf24 RCAF1 PNAS-11  Homo sapiens (Human)     137   \n",
       "7624                RABGAP1 HSPC094  Homo sapiens (Human)    1069   \n",
       "\n",
       "                                               Sequence  \n",
       "0     MAENPSLENHRIKSFKNKGRDVETMRRHRNEVTVELRKNKRDEHLL...  \n",
       "1     MADNEKLDNQRLKNFKNKGRDLETMRRQRNEVVVELRKNKRDEHLL...  \n",
       "2     MTTPGKENFRLKSYKNKSLNPDEMRRRREEEGLQLRKQKREEQLFK...  \n",
       "3     MSTNENANTPAARLHRFKNKGKDSTEMRRRRIEVNVELRKAKKDDQ...  \n",
       "4     MRIPVDASTSRRFTPPSTALSPGKMSEALPLGAPDAGAALAGKLRS...  \n",
       "...                                                 ...  \n",
       "7620  MSSFGYRTLTVALFTLICCPGSDEKVFEVHVRPKKLAVEPKGSLEV...  \n",
       "7621  MGATTMDQKSLWAGVVVLLLLQGGSAYKLVCYFTNWSQDRQEPGKF...  \n",
       "7622  MSLMVVSMACVGLFLVQRAGPHMGGQDKPFLSAWPSAVVPRGGHVT...  \n",
       "7623  MSGGRRKEEPPQPQLANGALKVSVWSKVLRSDAAWEDKDEFLDVIY...  \n",
       "7624  MDDKASVGKISVSSDSVSTLNSEDFVLVSRQGDETPSTNNGSDDEK...  \n",
       "\n",
       "[7625 rows x 9 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"Data/human_proteins_fixed.tab\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "80f17618-def6-4150-af21-92592f0a04a4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7bff71d4-7f2e-464f-8b4d-0d3a1a6730aa",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c9cff697-c377-466f-91df-bf54cb896a00",
   "metadata": {},
   "outputs": [],
   "source": [
    "genes = pd.read_csv(\"Data/human_genes.txt\", sep=\"\\t\", header=None).iloc[:, 0].values\n",
    "genes_all = pd.read_csv(\"Data/human_genes_all.txt\", sep=\"\\t\", header=None).iloc[:, 0].values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "258d95ca-42f6-4808-ba8c-db929883321f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(7482, 21340, 27024, 8788)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(genes), len(genes_all), len(virus_human), len(human_human)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c3dd425e-5887-43a7-b970-78f6a210a3a5",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'virus_human' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Input \u001b[0;32mIn [1]\u001b[0m, in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mvirus_human\u001b[49m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'virus_human' is not defined"
     ]
    }
   ],
   "source": [
    "virus_human"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ef874fe1-4835-4a24-b6e6-6d4e85771ff8",
   "metadata": {},
   "source": [
    "# Read JSON file to parse PFAM domain results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "f504a3c5-67d5-402b-930b-5c9058600b7b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Opening JSON file\n",
    "# f = open('Q09666.json')\n",
    " \n",
    "# # returns JSON object as\n",
    "# # a dictionary\n",
    "# data = json.load(f)\n",
    " \n",
    "# # Iterating through the json\n",
    "# # list\n",
    "# for data_dict in data['motifs']:\n",
    "#     print(data_dict['metadata']['type'])\n",
    " \n",
    "# # Closing file\n",
    "# f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "71f8035f-39bb-4a29-b593-bdc394aab490",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/skulkarni/Desktop/git/bst281_finalProject'"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8eb03c86-1780-4b7f-876d-7c97a214d83e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
